### PR TITLE
[Security][SecurityBundle] Backport type fixes

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | Backports from #43872
| License       | MIT
| Doc PR        | N/A

### CODE REVIEW

File: src/Symfony/Component/Security/Core/User/ChainUserProvider.php [hunk 1]
```diff
@@ -136,7 +136,7 @@ public function upgradePassword($user, string $newHashedPassword): void
             trigger_deprecation('symfony/security-core', '5.3', 'The "%s::upgradePassword()" method expects an instance of "%s" as first argument, the "%s" class should implement it.', PasswordUpgraderInterface::class, PasswordAuthenticatedUserInterface::class, get_debug_type($user));
 
             if (!$user instanceof UserInterface) {
-                throw new \TypeError(sprintf('The "%s::upgradePassword()" method expects an instance of "%s" as first argument, "%s" given.', PasswordAuthenticatedUserInterface::class, get_debug_type($user)));
+                throw new \TypeError(sprintf('The "%s::upgradePassword()" method expects an instance of "%s" as first argument, "%s" given.', static::class, PasswordAuthenticatedUserInterface::class, get_debug_type($user)));
```
Comment (derrabus) [inline src/Symfony/Component/Security/Core/User/ChainUserProvider.php]: Backported to 5.3 because this is a real bug: #43900

**Review by nicolas-grekas — APPROVED** [review]
(just please fix fabbot's report about deprecations)

**Comment by fabpot** [comment]
Can you also rebase?

File: src/Symfony/Component/Security/Core/Role/RoleHierarchy.php [hunk 2]
```diff
@@ -21,9 +21,6 @@ class RoleHierarchy implements RoleHierarchyInterface
     private $hierarchy;
     protected $map;
 
-    /**
-     * @param array $hierarchy An array defining the hierarchy
```
Comment (stof) [inline src/Symfony/Component/Security/Core/Role/RoleHierarchy.php:25]: could be described as `array<string, string[]>` instead

**Comment by derrabus** [comment]
ready

**Review by chalasr — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @derrabus.

**Pull request merged**